configure.ac

```diff
@@ -1247,6 +1247,16 @@ case "$host_os" in
 	;;
 esac
 
+dnl Visibility annotations...
+saved_CFLAGS="$CFLAGS"
+CFLAGS="$CFLAGS -fvisibility=hidden -Wall -Werror"
+AC_MSG_CHECKING(whether compiler supports visibility annotations)
+AC_COMPILE_IFELSE([AC_LANG_SOURCE([[void __attribute__((visibility("default"))) foo(void); void foo(void) {}]])],
+                  [CFLAGS="$saved_CFLAGS -fvisibility=hidden -D__LIBARCHIVE_ENABLE_VISIBILITY";
+                   AC_MSG_RESULT(yes)],
+                  [CFLAGS="$saved_CFLAGS"
+                   AC_MSG_RESULT(no)])
+
```

Comment:
Small request: can you port this to the cmake build? IIRC there are similar snippets to use as examples within.